decoder/stats.go

```diff
@@ -449,50 +451,43 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areas []geo.AreaName, now in
 }
 
 func updateRaidStats(old *Gym, new *Gym, areas []geo.AreaName) {
-	if len(areas) == 0 {
-		areas = []geo.AreaName{
-			{
-				Parent: "unmatched",
-				Name:   "unmatched",
-			},
-		}
-	}
-
-	areas = append(areas, geo.AreaName{
-		Parent: "world",
-		Name:   "world",
-	})
-
-	locked := false
-
-	// Loop though all areas
-	for i := 0; i < len(areas); i++ {
-		area := areas[i]
-
-		// Check if Raid has started/is active or RaidEndTimestamp has changed (Back-to-back raids)
-		if new.RaidPokemonId.ValueOrZero() > 0 &&
-			(old == nil || old.RaidPokemonId != new.RaidPokemonId || old.RaidEndTimestamp != new.RaidEndTimestamp) {
-
-			if !locked {
-				raidStatsLock.Lock()
-				locked = true
-			}
-
-			countStats := raidCount[area]
-			if countStats == nil {
-				countStats = make(map[int64]areaRaidCountDetail)
-				raidCount[area] = countStats
-			}
-
-			var countRaids = raidCount[area][new.RaidLevel.ValueOrZero()]
-			countRaids.count[new.RaidPokemonId.ValueOrZero()]++
-			raidCount[area][new.RaidLevel.ValueOrZero()] = countRaids
-		}
-	}
+    if len(areas) == 0 {
+        areas = []geo.AreaName{{Parent: "unmatched", Name: "unmatched"}}
+    }
+    areas = append(areas, geo.AreaName{Parent: "world", Name: "world"})
+
+    locked := false
+
+    for i := 0; i < len(areas); i++ {
+        area := areas[i]
+
+        if new.RaidPokemonId.ValueOrZero() > 0 &&
+            (old == nil || old.RaidPokemonId != new.RaidPokemonId || old.RaidEndTimestamp != new.RaidEndTimestamp) {
+
+            if !locked {
+                raidStatsLock.Lock()
+                locked = true
+            }
+
+            if raidCount[area] == nil {
+                raidCount[area] = make(map[int64]*areaRaidCountDetail)
+            }
+            countStats := raidCount[area]
+            raidLevel := new.RaidLevel.ValueOrZero()
+            if countStats[raidLevel] == nil {
+    countStats[raidLevel] = &areaRaidCountDetail{count: make(map[pokemonForm]int)}
+}
+pf := pokemonForm{
```

Comment:
The formatting here is wrong: I am not sure that you are using a go aware editor?
If you execute `go fmt` then it should reformat the file for you